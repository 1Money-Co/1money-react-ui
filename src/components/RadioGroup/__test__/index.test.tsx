import * as React from 'react';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { RadioGroup } from '../index';

configure({ adapter: new Adapter() });

describe('RadioGroup', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <RadioGroup items={[
        { name: 'US', key: 'USA', label: 'USA' },
        { name: 'CA', key: 'Canada', label: 'Canada', autoFocus: true },
        { name: 'MX', key: 'Mexico', label: 'Mexico' },
        { name: 'BR', key: 'Brazil', label: 'Brazil' },
      ]} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
