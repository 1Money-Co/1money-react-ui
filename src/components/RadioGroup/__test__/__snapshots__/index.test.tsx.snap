// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup renders correctly 1`] = `
LoadedCheerio {
  "0": <div
    class="usd1-react-ui-radiogroup-wrapper tw-flex"
  >
    <div
      class="usd1-react-ui-radiogroup-inner"
    >
      <div
        class="usd1-react-ui-radiogroup-inner-radio p-radiobutton p-component"
        data-p-checked="false"
        data-pc-name="radiobutton"
        data-pc-section="root"
      >
        <input
          class="p-radiobutton-input"
          data-pc-section="input"
          id="USA"
          name="US"
          type="radio"
        />
        <div
          class="p-radiobutton-box"
          data-pc-section="box"
        >
          <div
            class="p-radiobutton-icon"
            data-pc-section="icon"
          />
        </div>
      </div>
      <label
        for="USA"
      >
        USA
      </label>
    </div>
    <div
      class="usd1-react-ui-radiogroup-inner"
    >
      <div
        class="usd1-react-ui-radiogroup-inner-radio p-radiobutton p-component p-highlight"
        data-p-checked="true"
        data-pc-name="radiobutton"
        data-pc-section="root"
      >
        <input
          checked=""
          class="p-radiobutton-input"
          data-pc-section="input"
          id="Canada"
          name="CA"
          type="radio"
        />
        <div
          class="p-radiobutton-box"
          data-pc-section="box"
        >
          <div
            class="p-radiobutton-icon"
            data-pc-section="icon"
          />
        </div>
      </div>
      <label
        for="Canada"
      >
        Canada
      </label>
    </div>
    <div
      class="usd1-react-ui-radiogroup-inner"
    >
      <div
        class="usd1-react-ui-radiogroup-inner-radio p-radiobutton p-component"
        data-p-checked="false"
        data-pc-name="radiobutton"
        data-pc-section="root"
      >
        <input
          class="p-radiobutton-input"
          data-pc-section="input"
          id="Mexico"
          name="MX"
          type="radio"
        />
        <div
          class="p-radiobutton-box"
          data-pc-section="box"
        >
          <div
            class="p-radiobutton-icon"
            data-pc-section="icon"
          />
        </div>
      </div>
      <label
        for="Mexico"
      >
        Mexico
      </label>
    </div>
    <div
      class="usd1-react-ui-radiogroup-inner"
    >
      <div
        class="usd1-react-ui-radiogroup-inner-radio p-radiobutton p-component"
        data-p-checked="false"
        data-pc-name="radiobutton"
        data-pc-section="root"
      >
        <input
          class="p-radiobutton-input"
          data-pc-section="input"
          id="Brazil"
          name="BR"
          type="radio"
        />
        <div
          class="p-radiobutton-box"
          data-pc-section="box"
        >
          <div
            class="p-radiobutton-icon"
            data-pc-section="icon"
          />
        </div>
      </div>
      <label
        for="Brazil"
      >
        Brazil
      </label>
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
